PlagiarismSettingsForm.inc.php

```diff
@@ -31,37 +53,71 @@ function __construct($plugin, $contextId) {
 	/**
 	 * Initialize form data.
 	 */
-	function initData() {
-		list($username, $password) = $this->_plugin->getForcedCredentials();
-		$this->_data = array(
-                        'ithenticateUser' => $this->_plugin->getSetting($this->_contextId, 'ithenticateUser'),
-			'ithenticatePass' => $this->_plugin->getSetting($this->_contextId, 'ithenticatePass'),
-			'ithenticateForced' => !empty($username) && !empty($password)
-		);
+	public function initData() {
+		$this->_data = [
+			'ithenticateApiUrl' => $this->_plugin->getSetting($this->_context->getId(), 'ithenticateApiUrl'),
+			'ithenticateApiKey' => $this->_plugin->getSetting($this->_context->getId(), 'ithenticateApiKey'),
+			'ithenticateForced' => $this->_plugin->hasForcedCredentials(),
+		];
 	}
 
 	/**
 	 * Assign form data to user-submitted data.
 	 */
-	function readInputData() {
-                $this->readUserVars(array('ithenticateUser', 'ithenticatePass'));
+	public function readInputData() {
+		$this->readUserVars(['ithenticateApiUrl', 'ithenticateApiKey']);
 	}
 
 	/**
 	 * @copydoc Form::fetch()
 	 */
-	function fetch($request, $template = null, $display = false) {
+	public function fetch($request, $template = null, $display = false) {
 		$templateMgr = TemplateManager::getManager($request);
 		$templateMgr->assign('pluginName', $this->_plugin->getName());
 		return parent::fetch($request, $template, $display);
 	}
 
+	/**
+	 * @copydoc Form::validate()
+	 */
+	public function validate($callHooks = true) {
+		$this->addCheck(
```

Comment:
Checks should be added in the form constructor, not the call to `validate`.

Reply:
As we need to init the ithenticate service to test the given creds, using the `getData` in constructor does not work and pass null . well fixed it using `$request->getUserVar` and moved it to constructor .